src/Xamarin.Android.Build.Tasks/Utilities/MonoAndroidHelper.cs

```diff
@@ -325,6 +325,31 @@ public static bool HasMonoAndroidReference (MetadataReader reader)
 			return false;
 		}
 
+		public static bool HasResourceDesignerAssemblyReference (ITaskItem assembly)
+		{
+			if (!File.Exists (assembly.ItemSpec)) {
+				return false;
+			}
+			using var pe = new PEReader (File.OpenRead (assembly.ItemSpec));
+			var reader = pe.GetMetadataReader ();
+			if (!HasMonoAndroidReference (reader)) {
```

Comment:
Why would there be a requirement that assemblies which reference `_Microsoft.Android.Resource.Designer.dll` *also* reference `Mono.Android.dll`?

*Currently* that's the case because an assembly with a `Resource` type has static constructors which call `Android.Runtime.ResourceIdManager.UpdateIdValues()`.  This forces an assembly reference to `Mono.Android.dll`.

In our New World Order… Consider a [`dotnet new androidlib`](https://github.com/xamarin/xamarin-android/tree/main/src/Microsoft.Android.Templates/androidlib) project which contains a `Resources/values/strings.xml` file, and that's it.  The default `Class1.cs` doesn't subclass `Java.Lang.Object`, and the new "compatibility" Library Resource type won't call `Android.Runtime.ResourceIdManager.UpdateIdValues()`, so it is entirely plausible -- if untested -- for such an assembly to reference `_Microsoft.Android.Resource.Designer.dll` while *not* referencing `Mono.Android.dll`, because "just because" an assembly is referenced at compile time does *not* mean that the resulting assembly references the assembly.

Reply:
So the real question is does ` dotnet new androidlib` which only contains a resource and no code end up woth a Mono.Android.dll reference? I'll test this out.

this was mostly an optimization, if we can assume that only "android" assemblies will have a designer reference then we dont need to process EVERY assembly that is referenced only the "android" ones.

Reply:
While i like the idea of the optimization, I am not convinced that it's "safe" in the New World Order, *unless* the generated `__Microsoft.Android.Resource.Designer.cs` also has e.g. an assembly-level custom attribute which *forces* a reference to `Mono.Android.dll`.